plugin-manager/src/main/kotlin/com/itsaky/androidide/plugins/manager/services/IdeFileServiceImpl.kt

```
@@ -125,44 +100,98 @@ class IdeFileServiceImpl(
+    override fun writeStream(file: File, input: InputStream): Long {
+        ensureWritePermission()
+        ensurePathAllowed(file)
+
+        return try {
+            file.parentFile?.mkdirs()
+            BufferedInputStream(input).use { buffered ->
+                FileOutputStream(file).use { output ->
+                    copyInterruptible(buffered, output)
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)